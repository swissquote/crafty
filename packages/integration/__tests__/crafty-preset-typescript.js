/* global jest, describe, it, expect */
const testUtils = require("../utils");

// Add a high timeout because of https://github.com/facebook/jest/issues/8942
// Tests would be unreliable if they timeout >_<
jest.setTimeout(30000);

it("Lints TypeScript using the command", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-typescript/lints"
  );

  const result = await testUtils.run(["jsLint", "js/**/*.ts"], cwd);

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Lints TypeScript using the command, --preset recommended, --preset node", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-typescript/lints"
  );

  const result = await testUtils.run(
    ["jsLint", "js/**/*.ts", "--preset", "recommended", "--preset", "node"],
    cwd
  );

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Lints TypeScript using the command, --preset recommended", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-typescript/lints"
  );

  const result = await testUtils.run(
    ["jsLint", "js/**/*.ts", "--preset", "recommended"],
    cwd
  );

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeFalsy();
});

it("Lints TypeScript using the command, --preset format", async () => {
  const cwd = await testUtils.getCleanFixtures(
    "crafty-preset-typescript/lints"
  );

  const result = await testUtils.run(
    ["jsLint", "js/**/*.ts", "--preset", "format"],
    cwd
  );

  expect(result).toMatchSnapshot();

  // Files aren't generated on failed lint
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js")).toBeFalsy();
  expect(testUtils.exists(cwd, "dist/js/myBundle.min.js.map")).toBeFalsy();
});
