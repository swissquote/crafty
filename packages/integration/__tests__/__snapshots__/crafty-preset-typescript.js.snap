// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lints TypeScript using the command 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/Component.ts
  1:1  error  Replace \`⏎export·function·test(a:\` with \`export·function·test(a:·\`  @swissquote/swissquote/prettier/prettier
  3:1  error  Replace \`···return·a·+···\` with \`··return·a·+\`                      @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/SomeLibrary.ts
   1:1  error  Delete \`⏎\`                                            @swissquote/swissquote/prettier/prettier
   5:4  error  Insert \`;\`                                            @swissquote/swissquote/prettier/prettier
   9:1  error  Delete \`··\`                                           @swissquote/swissquote/prettier/prettier
  10:1  error  Delete \`····\`                                         @swissquote/swissquote/prettier/prettier
  11:3  error  Delete \`··\`                                           @swissquote/swissquote/prettier/prettier
  13:1  error  Delete \`··\`                                           @swissquote/swissquote/prettier/prettier
  14:3  error  Insert \`··\`                                           @swissquote/swissquote/prettier/prettier
  15:1  error  Replace \`····\` with \`··\`                              @swissquote/swissquote/prettier/prettier
  17:1  error  Replace \`····divide(a,·b)⏎···\` with \`··divide(a,·b)\`  @swissquote/swissquote/prettier/prettier
  19:1  error  Replace \`········\` with \`····\`                        @swissquote/swissquote/prettier/prettier
  20:1  error  Delete \`····\`                                         @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/script.ts
   1:1   error    Replace \`⏎import·{\` with \`import·{·\`         @swissquote/swissquote/prettier/prettier
   6:3   error    Replace \`··constructor·\` with \`constructor\`  @swissquote/swissquote/prettier/prettier
   7:5   warning  Unexpected console statement                 no-console
   8:2   error    Insert \`·\`                                   @swissquote/swissquote/prettier/prettier
  11:5   warning  Unexpected console statement                 no-console
  11:17  error    Replace \`·test(·\` with \`test(\`               @swissquote/swissquote/prettier/prettier
  13:16  error    Replace \`·(·)·=>⏎···\` with \`()·=>\`           @swissquote/swissquote/prettier/prettier
  17:7   warning  Unexpected console statement                 no-console
  18:6   error    Replace \`·,·300)\` with \`,·300);\`             @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/utils.ts
   6:9   error    Type string trivially inferred from a string literal, remove type annotation                                                                                                                  @swissquote/swissquote/@typescript-eslint/no-inferrable-types
   7:3   warning  Unexpected console statement                                                                                                                                                                  no-console
  29:10  error    React Hook \\"useSWROrig\\" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?  @swissquote/swissquote/react-hooks/rules-of-hooks
  34:25  error    Expected '===' and instead saw '=='                                                                                                                                                           eqeqeq
  35:16  error    Redundant use of \`await\` on a return value                                                                                                                                                    no-return-await

✖ 27 problems (23 errors, 4 warnings)
  20 errors and 0 warnings potentially fixable with the \`--fix\` option.

",
}
`;

exports[`Lints TypeScript using the command, --preset format 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/Component.ts
  1:1  error  Replace \`⏎export·function·test(a:\` with \`export·function·test(a:·\`  @swissquote/swissquote/prettier/prettier
  3:1  error  Replace \`···return·a·+···\` with \`··return·a·+\`                      @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/SomeLibrary.ts
   1:1  error  Delete \`⏎\`                                            @swissquote/swissquote/prettier/prettier
   5:4  error  Insert \`;\`                                            @swissquote/swissquote/prettier/prettier
   9:1  error  Delete \`··\`                                           @swissquote/swissquote/prettier/prettier
  10:1  error  Delete \`····\`                                         @swissquote/swissquote/prettier/prettier
  11:3  error  Delete \`··\`                                           @swissquote/swissquote/prettier/prettier
  13:1  error  Delete \`··\`                                           @swissquote/swissquote/prettier/prettier
  14:3  error  Insert \`··\`                                           @swissquote/swissquote/prettier/prettier
  15:1  error  Replace \`····\` with \`··\`                              @swissquote/swissquote/prettier/prettier
  17:1  error  Replace \`····divide(a,·b)⏎···\` with \`··divide(a,·b)\`  @swissquote/swissquote/prettier/prettier
  19:1  error  Replace \`········\` with \`····\`                        @swissquote/swissquote/prettier/prettier
  20:1  error  Delete \`····\`                                         @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/script.ts
   1:1   error  Replace \`⏎import·{\` with \`import·{·\`         @swissquote/swissquote/prettier/prettier
   6:3   error  Replace \`··constructor·\` with \`constructor\`  @swissquote/swissquote/prettier/prettier
   8:2   error  Insert \`·\`                                   @swissquote/swissquote/prettier/prettier
  11:17  error  Replace \`·test(·\` with \`test(\`               @swissquote/swissquote/prettier/prettier
  13:16  error  Replace \`·(·)·=>⏎···\` with \`()·=>\`           @swissquote/swissquote/prettier/prettier
  18:6   error  Replace \`·,·300)\` with \`,·300);\`             @swissquote/swissquote/prettier/prettier

✖ 19 problems (19 errors, 0 warnings)
  19 errors and 0 warnings potentially fixable with the \`--fix\` option.

",
}
`;

exports[`Lints TypeScript using the command, --preset recommended 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/Component.ts
  1:1  error  Replace \`⏎export·function·test(a:\` with \`export·function·test(a:·\`  @swissquote/swissquote/prettier/prettier
  3:1  error  Replace \`···return·a·+···\` with \`··return·a·+\`                      @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/SomeLibrary.ts
   1:1  error  Delete \`⏎\`                                            @swissquote/swissquote/prettier/prettier
   5:4  error  Insert \`;\`                                            @swissquote/swissquote/prettier/prettier
   9:1  error  Delete \`··\`                                           @swissquote/swissquote/prettier/prettier
  10:1  error  Delete \`····\`                                         @swissquote/swissquote/prettier/prettier
  11:3  error  Delete \`··\`                                           @swissquote/swissquote/prettier/prettier
  13:1  error  Delete \`··\`                                           @swissquote/swissquote/prettier/prettier
  14:3  error  Insert \`··\`                                           @swissquote/swissquote/prettier/prettier
  15:1  error  Replace \`····\` with \`··\`                              @swissquote/swissquote/prettier/prettier
  17:1  error  Replace \`····divide(a,·b)⏎···\` with \`··divide(a,·b)\`  @swissquote/swissquote/prettier/prettier
  19:1  error  Replace \`········\` with \`····\`                        @swissquote/swissquote/prettier/prettier
  20:1  error  Delete \`····\`                                         @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/script.ts
   1:1   error    Replace \`⏎import·{\` with \`import·{·\`         @swissquote/swissquote/prettier/prettier
   6:3   error    Replace \`··constructor·\` with \`constructor\`  @swissquote/swissquote/prettier/prettier
   7:5   warning  Unexpected console statement                 no-console
   8:2   error    Insert \`·\`                                   @swissquote/swissquote/prettier/prettier
  11:5   warning  Unexpected console statement                 no-console
  11:17  error    Replace \`·test(·\` with \`test(\`               @swissquote/swissquote/prettier/prettier
  13:16  error    Replace \`·(·)·=>⏎···\` with \`()·=>\`           @swissquote/swissquote/prettier/prettier
  17:7   warning  Unexpected console statement                 no-console
  18:6   error    Replace \`·,·300)\` with \`,·300);\`             @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/utils.ts
   6:9   error    Type string trivially inferred from a string literal, remove type annotation                                                                                                                  @swissquote/swissquote/@typescript-eslint/no-inferrable-types
   7:3   warning  Unexpected console statement                                                                                                                                                                  no-console
  29:10  error    React Hook \\"useSWROrig\\" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?  @swissquote/swissquote/react-hooks/rules-of-hooks
  34:25  error    Expected '===' and instead saw '=='                                                                                                                                                           eqeqeq
  35:16  error    Redundant use of \`await\` on a return value                                                                                                                                                    no-return-await

✖ 27 problems (23 errors, 4 warnings)
  20 errors and 0 warnings potentially fixable with the \`--fix\` option.

",
}
`;

exports[`Lints TypeScript using the command, --preset recommended, --preset node 1`] = `
Object {
  "status": 1,
  "stdall": "
[__:__:__] Starting Crafty __version__...

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/Component.ts
  1:1  error  Replace \`⏎export·function·test(a:\` with \`export·function·test(a:·\`  @swissquote/swissquote/prettier/prettier
  3:1  error  Replace \`···return·a·+···\` with \`··return·a·+\`                      @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/SomeLibrary.ts
   1:1  error  Delete \`⏎\`                                            @swissquote/swissquote/prettier/prettier
   5:4  error  Insert \`;\`                                            @swissquote/swissquote/prettier/prettier
   9:1  error  Delete \`··\`                                           @swissquote/swissquote/prettier/prettier
  10:1  error  Delete \`····\`                                         @swissquote/swissquote/prettier/prettier
  11:3  error  Delete \`··\`                                           @swissquote/swissquote/prettier/prettier
  13:1  error  Delete \`··\`                                           @swissquote/swissquote/prettier/prettier
  14:3  error  Insert \`··\`                                           @swissquote/swissquote/prettier/prettier
  15:1  error  Replace \`····\` with \`··\`                              @swissquote/swissquote/prettier/prettier
  17:1  error  Replace \`····divide(a,·b)⏎···\` with \`··divide(a,·b)\`  @swissquote/swissquote/prettier/prettier
  19:1  error  Replace \`········\` with \`····\`                        @swissquote/swissquote/prettier/prettier
  20:1  error  Delete \`····\`                                         @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/script.ts
   1:1   error  Replace \`⏎import·{\` with \`import·{·\`         @swissquote/swissquote/prettier/prettier
   6:3   error  Replace \`··constructor·\` with \`constructor\`  @swissquote/swissquote/prettier/prettier
   8:2   error  Insert \`·\`                                   @swissquote/swissquote/prettier/prettier
  11:17  error  Replace \`·test(·\` with \`test(\`               @swissquote/swissquote/prettier/prettier
  13:16  error  Replace \`·(·)·=>⏎···\` with \`()·=>\`           @swissquote/swissquote/prettier/prettier
  18:6   error  Replace \`·,·300)\` with \`,·300);\`             @swissquote/swissquote/prettier/prettier

__PATH__/packages/integration/fixtures/crafty-preset-typescript/lints/js/utils.ts
   6:9   error  Type string trivially inferred from a string literal, remove type annotation                                                                                                                  @swissquote/swissquote/@typescript-eslint/no-inferrable-types
  29:10  error  React Hook \\"useSWROrig\\" is called conditionally. React Hooks must be called in the exact same order in every component render. Did you accidentally call a React Hook after an early return?  @swissquote/swissquote/react-hooks/rules-of-hooks
  34:25  error  Expected '===' and instead saw '=='                                                                                                                                                           eqeqeq
  35:16  error  Redundant use of \`await\` on a return value                                                                                                                                                    no-return-await

✖ 23 problems (23 errors, 0 warnings)
  20 errors and 0 warnings potentially fixable with the \`--fix\` option.

",
}
`;
